solace-spring-cloud-starters/solace-spring-cloud-stream-starter/README.adoc

```diff
@@ -757,6 +771,127 @@ Specifies that the dynamic destination is a queue
+----
+
+=== Behavior
+
+==== Consumer Side (Solace to Spring)
+When the binder consumes messages from Solace and converts them to Spring Cloud Stream messages:
+
+1. **Header Mapping Resolution**: The binder examines each Solace user property and checks if it has a corresponding mapping in the `headerNameMapping` configuration
+2. **Mapped Properties**: User properties with configured mappings are converted to Spring message headers using the mapped header name (e.g., `timestamp` user property becomes `original-timestamp` header)
+3. **Unmapped Properties**: User properties without mappings are converted to Spring message headers using their original property names, in which case it may be overwritten if it is Spring Integration reserved header (e.g. id, timestamp etc)
+4. **Reserved Header Preservation**: This allows access to original Solace message metadata that would otherwise be overwritten by Spring Cloud Stream's reserved headers during message processing
+
+==== Producer Side (Spring to Solace)
+When the binder publishes Spring Cloud Stream messages to Solace:
+
+1. **Header Mapping Lookup**: The binder examines each Spring message header and checks if it has a corresponding mapping in the `headerNameMapping` configuration
+2. **Mapped Headers**: Headers with configured mappings are converted to Solace user properties using the mapped property name (e.g., `original-timestamp` header becomes `timestamp` user property)
+3. **Unmapped Headers**: Headers without mappings are converted to Solace user properties using their original header names
+4. **Reserved Header Handling**: This prevents Spring Cloud Stream reserved headers from overwriting important application-specific metadata when publishing to Solace
+
+=== Backward Compatibility
+
+This feature is fully backward compatible:
+
+* If no `headerNameMapping` is configured, the binder behaves exactly as before
+* Existing applications will continue to work without any changes
+
+=== Example Use Cases
+
+==== Preserving Reserved Headers
+When you need to retain the original values of Spring Cloud Stream reserved headers:
+
+[source,yaml]
+----
+spring:
+  cloud:
+    stream:
+      solace:
+        bindings:
+          data-processor-in-0:
+            consumer:
+              headerNameMapping:
+                timestamp: original-timestamp        #Preserve publisher application set timestamp
+                id: original-id             #Preserve publisher application set id
+                correlationId: original-correlationId  #Preserve publisher application set correlation ID
+----
+
+==== Cross-System Integration
+When integrating with legacy systems that expect specific property names:
+
+[source,yaml]
+----
+spring:
+  cloud:
+    stream:
+      solace:
+        bindings:
+          legacy-system-out-0:
+            producer:
+              headerNameMapping:
+                id: MSG_ID
+                timestamp: TS
+                spring-source: SRC_SYS
+----
+
```

Comment:
do you think this is too verbose? It seems to be saying the same thing as earlier in the section?